### PR TITLE
gtk3: restore x11 support on darwin

### LINKED ISSUE #95946: peek: fails to build on macOS
**Describe the bug**
peek fails to build on macOS

**To Reproduce**
Clone `nixpkgs` then `nix-build -A peek`

**Notify maintainers**
@puffnfresh @worldofpeace 

**Log**
```
keybinder-3.0 v0.3.2

Prefix............... : /nix/store/8zd2sdjmbpw9c9rsk88g1nxjvjbhw0w9-keybinder3-0.3.2
GTK version ......... : gtk+-3.0
libkeybinder ........ : yes
gobject-introspection : yes
gtk-doc ............. : no

Now type make to compile
Then su to root and type: make install

building
build flags: SHELL=/nix/store/saggdc3mgg1h6fs05r9rv5isypqwzby8-bash-4.4-p23/bin/bash
Making all in libkeybinder
make[1]: Entering directory '/private/var/folders/s1/sm71lqk553d1_f8drd3tz_5r0000gn/T/nix-build-keybinder3-0.3.2.drv-0/source/libkeybinder'
  CC     bind.lo
bind.c:30:10: fatal error: 'gdk/gdkx.h' file not found
#include <gdk/gdkx.h>
         ^~~~~~~~~~~~
1 error generated.
make[1]: *** [Makefile:424: bind.lo] Error 1
make[1]: Leaving directory '/private/var/folders/s1/sm71lqk553d1_f8drd3tz_5r0000gn/T/nix-build-keybinder3-0.3.2.drv-0/source/libkeybinder'
make: *** [Makefile:359: all-recursive] Error 1
builder for '/nix/store/kpd2gm2yfk840b18yi9yfxhd66klpbc2-keybinder3-0.3.2.drv' failed with exit code 2
cannot build derivation '/nix/store/vz8jh6pj5h6sd9w2s19s85ignk10y5ba-peek-1.5.1.drv': 1 dependencies couldn't be built
error: build of '/nix/store/vz8jh6pj5h6sd9w2s19s85ignk10y5ba-peek-1.5.1.drv' failed
```

### LINKED ISSUE #132259: darwin: gtk3-x11 is built without X11 support
### Describe the bug

`gtk3-x11` is built without X11 support on darwin because the Meson build file disables it.

### Steps To Reproduce

The GTK build system was changed to Meson by 638c39e65bce5adcea7b7bebf3586c7ee371563f, and it removed the configure flags that enabled X11 on darwin:

``` diff
-  configureFlags = optional stdenv.isDarwin [
-    "--disable-debug"
-    "--disable-dependency-tracking"
-    "--disable-glibtest"
-  ] ++ optional (stdenv.isDarwin && !x11Support)
-    "--enable-quartz-backend"
-    ++ optional x11Support [
-    "--enable-x11-backend"
-  ] ++ optional waylandSupport [
-    "--enable-wayland-backend"
-  ];
```

But the `meson.build` file of GTK3 disables X11 support on darwin, and [it is harcoded](https://gitlab.gnome.org/GNOME/gtk/-/blob/gtk-3-24/meson.build#L155-160):

``` meson
if os_darwin
  wayland_enabled = false
  x11_enabled = false
else
  quartz_enabled = false
endif 
```

### Expected behavior

`gtk3-x11` should provide X11 support on darwin.

### Additional context

I was trying to fix `pympress` on darwin and I found that `libcanberra-gtk3` is marked as broken (806d814516c14853e7dae37eb9a59c96db557786) because of a missing GKT/X11 header.

It seems that it is the same problem that motivated #82097 to change Vim to use "gtk2" (same missing header `gdk/gdkx.h`).

### Notify maintainers

@hedning @jtojnar @dasj19 @maxeaubrey

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 20.4.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.3.12`
 - channels(smancill): `"nixpkgs-21.05pre293250.c8569c32bac, unstable-21.11pre304227.ecaf3da9340"`
 - nixpkgs: `/Users/smancill/.nix-defexpr/channels/nixpkgs`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- gtk3-x11
- libcanberra-gtk3
- vim_configurable
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by Artturin**
###### Motivation for this change
Might closes #95946
Closes https://github.com/NixOS/nixpkgs/issues/132259

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 132239` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>quodlibet-full</li>
    <li>quodlibet-xine-full</li>
    <li>synapse</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>keybinder3</li>
  </ul>
</details>

**Comment by Artturin** [comment]
same error?

**Comment by siraben** [comment]
```
keybinder-3.0 v0.3.2

Prefix............... : /nix/store/f2312gwkrida11m5ks5klxxh2463hpjv-keybinder3-0.3.2
GTK version ......... : gtk+-3.0
libkeybinder ........ : yes
gobject-introspection : yes
gtk-doc ............. : no

Now type make to compile
Then su to root and type: make install

@nix { "action": "setPhase", "phase": "buildPhase" }
building
build flags: SHELL=/nix/store/715y5lnw12vdcc3hh9sc9vbkdi6c83x6-bash-4.4-p23/bin/bash
Making all in libkeybinder
make[1]: Entering directory '/private/tmp/nix-build-keybinder3-0.3.2.drv-0/source/libkeybinder'
  CC     bind.lo
bind.c:30:10: fatal error: 'gdk/gdkx.h' file not found
#include <gdk/gdkx.h>
         ^~~~~~~~~~~~
1 error generated.
make[1]: *** [Makefile:424: bind.lo] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-keybinder3-0.3.2.drv-0/source/libkeybinder'
make: *** [Makefile:359: all-recursive] Error 1
```

**Comment by Artturin** [comment]
thats after the force push or before?

**Comment by Artturin** [comment]
Looks like this could be related https://github.com/NixOS/nixpkgs/issues/132259

**Comment by siraben** [comment]
> thats after the force push or before?

before. I'm running another review now.

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 132239` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>keybinder3</li>
  </ul>
</details>

**Comment by siraben** [comment]
Still same issue.

**Comment by jtojnar** [comment]
Unless something changed, explicitly enabling X11 will not do anything as https://gitlab.gnome.org/GNOME/gtk/-/commit/9109efb19e4bc92a81ad972f938856c6d6bf01f8 disables it unconditionally on Darwin.

**Comment by Artturin** [comment]
> Unless something changed, explicitly enabling X11 will not do anything as [gitlab.gnome.org/GNOME/gtk/-/commit/9109efb19e4bc92a81ad972f938856c6d6bf01f8](https://gitlab.gnome.org/GNOME/gtk/-/commit/9109efb19e4bc92a81ad972f938856c6d6bf01f8) disables it unconditionally on Darwin.

removed the disabling

used sed so that only the first occurrence of `x11_enabled = false` will be removed

File: pkgs/development/libraries/gtk/3.x.nix [hunk 1]
```diff
@@ -159,6 +161,9 @@ stdenv.mkDerivation rec {
   NIX_CFLAGS_COMPILE = "-DG_ENABLE_DEBUG -DG_DISABLE_CAST_CHECKS";
 
   postPatch = ''
+    # See https://github.com/NixOS/nixpkgs/issues/132259
+    sed -i "0,/x11_enabled = false/s///" meson.build
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/gtk/3.x.nix:165]: Can we use substituteInPlace to at least log a warning when this fails?

Comment (Artturin) [reply pkgs/development/libraries/gtk/3.x.nix:165]: alright. substituteInPlace will also remove the flag for windows but that probably wont matter https://gitlab.gnome.org/GNOME/gtk/-/blob/gtk-3-24/meson.build#L164

**Pull request merged**